### ISSUE TITLE
dev guide: e2e tests error "value must be a valid database engine."

### ISSUE THREAD

**Issue by avodonosov** [metabase/metabase#35306]
### Describe the bug

When running e2e tests as suggested in dev guide, with `yarn run test-cypress-run`, if fails like below. The dev guide is missing something I suppose:

```
$ yarn run test-cypress-run
yarn run v1.22.19
$ node ./e2e/runner/run_cypress_tests.js
Metabase version info
Running e2e test runner with this build:
tag=v0.47.6-SNAPSHOT
hash=e7ee12e
branch=heads/v1.47.5
date=2023-10-20
If that version seems too old, please run `./bin/build.sh :steps '[:version :uberjar]'`.
Starting backend
Waiting for backend (host=http://localhost:4000, dbFile=/tmp/metabase-test-47755.db)....................................................................................................
Backend ready host=http://localhost:4000, dbFile=/tmp/metabase-test-47755.db
Generating snapshots
DeploySentinel Cypress Debugger: [v0.8.8] is disabled (key not found)

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        12.8.1                                                                         │
  │ Browser:        Electron 106 (headless)                                                        │
  │ Node Version:   v20.9.0 (/usr/bin/node)                                                        │
  │ Specs:          1 found (default.cy.snap.js)                                                   │
  │ Searched:       e2e/snapshot-creators/**/*.cy.snap.js                                          │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  default.cy.snap.js                                                              (1 of 1)
Warning: The following browser launch options were provided but are not supported by electron

 - args


  snapshots
    default
      ✓ default (8632ms)
    withSqlite
      1) withSqlite


  1 passing (11s)
  1 failing

  1) snapshots
       withSqlite
         withSqlite:
     CypressError: `cy.request()` failed on:
                                                                                                                                      
http://localhost:4000/api/database                                                                                                    
                                                                                                                                      
The response we received from your web server was:                                                                                    
                                                                                                                                      
  > 400: Bad Request                                                                                                                  
                                                                                                                                      
This was considered a failure because the status code was not `2xx` or `3xx`.                                                         
                                                                                                                                      
If you do not want status codes to cause failures pass the option: `failOnStatusCode: false`                                          
                                                                                                                                      
-----------------------------------------------------------                                                                           
                                                                                                                                      
The request we sent was:                                                                                                              
                                                                                                                                      
Method: POST                                                                                                                          
URL: http://localhost:4000/api/database                                                                                               
Headers: {                                                                                                                            
  "Connection": "keep-alive",                                                                                                         
  "user-agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/12.8.1 Chrome/106.0.5249.51 Electron/21.0.0 Safari/537.36",                                                                                                                 
  "accept": "*/*",                                                                                                                    
  "cookie": "metabase.TIMEOUT=alive; metabase.DEVICE=3cb1d528-ccfd-4cc3-8be4-09da91649b4d; metabase.SESSION=de45b920-3cb5-4212-b4f0-cf9d9344ffe8",                                                                                                                          
  "accept-encoding": "gzip, deflate",                                                                                                 
  "content-type": "application/json",                                                                                                 
  "content-length": 355                                                                                                               
}                                                                                                                                     
Body: {"engine":"sqlite","name":"sqlite","details":{"db":"./resources/sqlite-fixture.db"},"auto_run_queries":true,"is_full_sync":true,"schedules":{"cache_field_values":{"schedule_day":null,"schedule_frame":null,"schedule_hour":0,"schedule_type":"daily"},"metadata_sync":{"schedule_day":null,"schedule_frame":null,"schedule_hour":null,"schedule_type":"hourly"}}}                                         
                                                                                                                                      
-----------------------------------------------------------                                                                           
                                                                                                                                      
The response we got was:                                                                                                              
                                                                                                                                      
Status: 400 - Bad Request                                                                                                             
Headers: {                                                                                                                            
  "x-frame-options": "DENY",                                                                                                          
  "x-xss-protection": "1; mode=block",                                                                                                
  "last-modified": "Wed, 1 Nov 2023 04:04:28 -0700",                                                                                  
  "strict-transport-security": "max-age=31536000",                                                                                    
  "set-cookie": [                                                                                                                     
    "metabase.TIMEOUT=alive;SameSite=Lax;Path=/;Max-Age=1209600"                                                                      
  ],                                                                                                                                  
  "x-permitted-cross-domain-policies": "none",                                                                                        
  "cache-control": "max-age=0, no-cache, must-revalidate, proxy-revalidate",                                                          
  "x-content-type-options": "nosniff",                                                                                                
  "content-security-policy": "default-src 'none'; script-src 'self' 'unsafe-eval' https://maps.google.com https://accounts.google.com    'sha256-K2AkR/jTLsGV8PyzWha7/ey1iaD9c5jWRYwa++ZlMZc=' 'sha256-ib2/2v5zC6gGM6Ety7iYgBUvpy/caRX9xV/pzzV7hf0=' 'sha256-isH538cVBUY8IMlGYGbWtBwr+cGqkc4mN6nLcA7lUjE='; child-src 'self' https://accounts.google.com; style-src 'self' 'unsafe-inline' https://accounts.google.com; font-src *; img-src * 'self' data:; connect-src 'self' https://accounts.google.com metabase.us10.list-manage.com   ; manifest-src 'self';  frame-ancestors 'none';",                                                                                                  
  "content-type": "application/json;charset=utf-8",                                                                                   
  "expires": "Tue, 03 Jul 2001 06:00:00 GMT",                                                                                         
  "content-length": "62",                                                                                                             
  "server": "Jetty(11.0.14)"                                                                                                          
}                                                                                                                                     
Body: {                                                                                                                               
  "errors": {                                                                                                                         
    "engine": "value must be a valid database engine."                                                                                
  }                                                                                                                                   
}                                                                                                                                     
                                                                                                                                      
                                                                                                                                      
https://on.cypress.io/request                                                                                                         
      at <unknown> (http://localhost:4000/__cypress/runner/cypress_runner.js:139729:80)                                               
      at tryCatcher (http://localhost:4000/__cypress/runner/cypress_runner.js:8914:23)                                                
      at Promise._settlePromiseFromHandler (http://localhost:4000/__cypress/runner/cypress_runner.js:6849:31)                         
      at Promise._settlePromise (http://localhost:4000/__cypress/runner/cypress_runner.js:6906:18)                                    
      at Promise._settlePromise0 (http://localhost:4000/__cypress/runner/cypress_runner.js:6951:10)                                   
      at Promise._settlePromises (http://localhost:4000/__cypress/runner/cypress_runner.js:7031:18)                                   
      at _drainQueueStep (http://localhost:4000/__cypress/runner/cypress_runner.js:3621:12)                                           
      at _drainQueue (http://localhost:4000/__cypress/runner/cypress_runner.js:3614:9)                                                
      at ../../node_modules/bluebird/js/release/async.js.Async._drainQueues (http://localhost:4000/__cypress/runner/cypress_runner.js:3630:5)                                                                                                                               
      at Async.drainQueues (http://localhost:4000/__cypress/runner/cypress_runner.js:3500:14)                                         
  From Your Spec Code:                                                                                                                
      at Context.eval (http://localhost:4000/__cypress/tests?p=e2e/snapshot-creators/default.cy.snap.js:43174:12)                     
                                                                                                                                      



  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        2                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     10 seconds                                                                       │
  │ Spec Ran:     default.cy.snap.js                                                               │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  /home/anton/my/tmp/test-my-clojure/metabase/cypress/screenshots/default.cy.snap.     (1280x720)
     js/snapshots -- withSqlite -- withSqlite (failed).png                                          


  (Video)

  -  Started processing:  Compressing to 32 CRF                                                     
  -  Finished processing: 3 seconds                                                  

  -  Video output: /home/anton/my/tmp/test-my-clojure/metabase/cypress/videos/default.cy.snap.js.mp4


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✖  default.cy.snap.js                       00:10        2        1        1        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  1 of 1 failed (100%)                     00:10        2        1        1        -        -  

Cleaning up...
Stopped backend (host=http://localhost:4000, dbFile=/tmp/metabase-test-47755.db)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### To Reproduce

Run e2e tests as suggested in dev guide, with `yarn run test-cypress-run`


### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

```JSON
checked out the tag v1.47.5

uname -a:
Linux mymachine 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

java -version:
openjdk version "16" 2021-03-16
OpenJDK Runtime Environment (build 16+36-2231)
OpenJDK 64-Bit Server VM (build 16+36-2231, mixed mode, sharing)
```


### Severity

-

### Additional context

_No response_

**Comment by ranquild** [comment]
@avodonosov run `./bin/build-drivers.sh` before running cypress. It's mentioned in dev docs.

**Comment by avodonosov** [comment]
@ranquild, 

I managed to have e2e tests running yesterday. I've tried various commands mentioned in the guide, including ./bin/build-drivrs.sh. In the end I was not sure how I made it work. IMHO the dev guide is somewhat unclear for a new developer - I suspected drivers, but why one snapshot test passed (the default) - one driver included by default and the other is not? Another question that left impression of uncertainty for me - how dependencies between various build and test tasks are handled. Should I know for a specific task all the prerequisite tasks and run them manually or build scripts take care of the dependent tasks (like in maven, `mvn package` implies `compile` and `test` and runs them automatically). 

Anyways, thank you for the response.
